### PR TITLE
Add support for monitoring specific processes

### PR DESCRIPTION
**Pull request by Porges**
Add a new commandline parameter which is a comma-separated list of process names to monitor. Stats for CPU and memory of these processes will be sent alongside the other metrics.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -20,6 +21,11 @@ func main() {
 		appInsightsKey = os.Args[3]
 	}
 
+	processNames := []string {}
+	if len(os.Args) > 4 {
+		processNames = strings.Split(os.Args[4], ",")
```
[the remark this answers is not in the document]
Comment (Porges) [reply main.go]: Added trim on input so whitespace is removed.

File: pkg/processes.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+				// process might have disappeared
+				continue
+			}
+
+			// check if we should include it
+			if !contains(processNames, name) {
+				continue
+			}
+
+			cpuPercent, err := p.CPUPercent()
```
Comment (timotheeguerin) [inline pkg/processes.go]: Does this work fine on windows? The main CPU usage has been implemented wrongly on windows(Or at least was when I made it, might have been fixed) I had to call wmi directly

File: main.go [hunk 3]
```diff
@@ -20,6 +21,15 @@ func main() {
 		appInsightsKey = os.Args[3]
 	}
 
+	processNames := []string {}
+	if len(os.Args) > 4 {
```
[the remark this answers is not in the document]
Comment (timotheeguerin) [reply main.go]: I don't think it is working with the one liner scripts like that

File: pkg/processes.go (hunk 2 above)
Comment (Porges) [reply pkg/processes.go]: That's annoying...

File: main.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Porges) [reply main.go]: Added `AZ_BATCH_MONITOR_PROCESSES`

File: pkg/processes.go (hunk 2 above)
Comment (timotheeguerin) [reply pkg/processes.go]: I saw a PR on the repo that might have fixed it recently so it might just be working fine

**Comment by timotheeguerin** [comment]
Can you also add docs to the readme on how to use this setting

**Comment by Porges** [comment]
Updated docs.

**Comment by timotheeguerin** [comment]
On windows you also need to update the `run-windows.ps1` because of the way it starts background tasks environment variables are not passed through

**Comment by Porges** [comment]
@timotheeguerin yes, that's why I wrote "this will require modifying the appropriate run script". Do you think I should be more explicit?

**Comment by timotheeguerin** [comment]
Oh I see, any reason why you wouldn't change it here? Feel like this is an unnecessary step for users

**Comment by Porges** [comment]
Right, easy enough 🙂

**Pull request merged**